# Traversability map filter parameters
traversability_map_filters:
  - name: surfaceNormalsFilter
    type: gridMapFilters/NormalVectorsFilter
    params:
      input_layer: elevation
      output_layers_prefix: surface_normal_
      radius: 0.05
      normal_vector_positive_axis: z
  - name: slopeFilter
    type: traversabilityFilters/SlopeFilter
    params:
      map_type: traversability_slope
      critical_value: 1.0
  - name: stepFilter
    type: traversabilityFilters/StepFilter
    params:
      map_type: traversability_step
      critical_value: 0.12
      first_window_radius: 0.04
      second_window_radius: 0.04
      critical_cell_number: 4
  - name: roughnessFilter
    type: traversabilityFilters/RoughnessFilter
    params:
      map_type: traversability_roughness
      critical_value: 0.05
      estimation_radius: 0.05
  - name: weightedSumFilter
    type: gridMapFilters/MathExpressionFilter
    params:
      output_layer: traversability
      expression: (1.0 / 3.0) * (traversability_slope + traversability_step + traversability_roughness)
  - name: deletionFilter
    type: gridMapFilters/DeletionFilter
    params:
      layers: [surface_normal_x, surface_normal_y, surface_normal_z]
